polaris-react/src/components/IndexTable/components/Row/Row.tsx

```diff
@@ -21,6 +22,8 @@ export interface RowProps {
   children: React.ReactNode;
   /** A unique identifier for the row */
   id: string;
+  /** Whether the row should hide the selectable checkbox while the parent IndexTable is selectable */
+  hideSelectable?: boolean;
```

Comment:
Is the desired usecase to remove the "select all" checkbox for subheadings?

If so, why? Can you link to some design / UX documentation on this feature request?

Assuming there's valid justification, it might make sense to link this prop directly to the subheading (if that's the usecase) - I can't think of another valid case where we'd want a mixture of enabled/disabled checkboxes in a table (which this generic prop enables).

Reply:
No, this is not our intended use case - it was just a scenario that made sense and conveniently had example code.

We need this to avoid showing check boxes for nested resources that do not have bulk actions like their parents. Additionally, since child resources can be dynamically loaded in, we have rows that do things like "load more" and "collapse" that ideally wouldnt have a checkbox either. [This issue](https://github.com/Shopify/core-issues/issues/69550) has a design of the specific case.

Reply:
(The use case isn't subheaders, its the opposite--the need is to make nested child rows not selectable)

Reply:
That makes sense, thanks for the added context!